pkgs/development/r-modules/generate-r-packages.R

```diff
@@ -71,44 +72,28 @@ formatPackage <- function(name, version, sha256, depends, imports, linkingTo) {
     depends <- depends[depends %in% knownPackages]
     depends <- lapply(depends, escapeName)
     depends <- paste(depends)
-    depends <- paste(sort(unique(depends)), collapse=" ")
-    paste0("  ", attr, " = derive2 { name=\"", name, "\"; version=\"", version, "\"; sha256=\"", sha256, "\"; depends=[", depends, "]; };")
+    depends <- sort(unique(depends))
+    list(version=unbox(version), sha256=unbox(sha256), depends=depends)
 }
 
 clusterExport(cl, c("nixPrefetch","readFormatted", "mirrorUrl", "mirrorType", "knownPackages"))
-
 pkgs <- pkgs[order(Package)]
 
 write(paste("updating", mirrorType, "packages"), stderr())
 pkgs$sha256 <- parApply(cl, pkgs, 1, function(p) nixPrefetch(p[1], p[2]))
-nix <- apply(pkgs, 1, function(p) formatPackage(p[1], p[2], p[18], p[4], p[5], p[6]))
-write("done", stderr())
 
 # Mark deleted packages as broken
-setkey(readFormatted, V2)
-markBroken <- function(name) {
-  str <- paste0(readFormatted[name], collapse='"')
-  if(sum(grep("broken = true;", str)))
-    return(str)
-  write(paste("marked", name, "as broken"), stderr())
-  gsub("};$", "broken = true; };", str)
-}
-broken <- lapply(setdiff(readFormatted[[2]], pkgs[[1]]), markBroken)
+brokenPkgs <- lapply(readFormatted[setdiff(names(readFormatted), pkgs$Packages)], function(p)
```

Comment:
is `pkgs$Packages` a typo?

Reply:
This is probably the culprit of the duplicate keys

Reply:
Though, `names(readFormatted)` is alread escaped, so we should probably use `escapedNames` here

Reply:
Pushed a fix